### DISCUSSION TITLE
How to get YAML file using APIs?

### DISCUSSION THREAD

**Discussion by sehrishnaseer** [kubeapps/kubeapps#3966] [Q&A] [answered]
How to get YAML file using APIs while creatingInstalledPackage or creatingRelease, where we can set resources and get all other info, which is shown in Installation values, after deploying the app? Any help would be appreciated!

**Comment by antgamdia** [comment, marked as answer]
You need to query the `GetAvailablePackageDetail` operation and retrieve the `defaultValues` field.
If using the REST API, your API call will be similar to:

```bash
curl -X 'GET' \
  'https://kubeapps.20.67.191.243.sslip.io/apis/core/packages/v1alpha1/availablepackages/plugin/helm.packages/v1alpha1/c/default/ns/kubeapps/bitnami%2Fairflow?pkgVersion=11.1.10' \
  -H 'accept: application/json'
```

![image](https://user-images.githubusercontent.com/11535726/146402944-869ab144-65de-4c9e-96c5-c2868ec76bc4.png)

Similarly, if interesting in getting the values of an already installed application, you'll need to query the `GetInstalledPackageDetail` operation instead.

**Reply by sehrishnaseer** [reply]
Thank you so much, it was really helpful
